.github/workflows/build-deploy-frontend.yml

```diff
@@ -24,55 +19,34 @@ permissions:
   contents: read
 
 jobs:
-  build:
-    name: Build
+  build-frontend:
     runs-on: ubuntu-latest
-    defaults:
-        run:
-          working-directory: frontend
-    
     steps:
-      - name: Checkout code
-        uses: actions/checkout@v4
-      - name: Install Node.js
-        uses: actions/setup-node@v4
+      - uses: actions/checkout@v4
+      - uses: ./.github/actions/build-frontend
+        name: Build frontend
         with:
-          node-version: '20'  
-      - name: Install NPM packages
-        run: npm ci
-      - name: Build project
-        run: VITE_API_URL='https://reportvision-ocr-${{ inputs.deploy-env }}.azurewebsites.net/' npm run build
-      - name: Run unit tests
-        run: npm run test
-      - name: Create client build archive
-        shell: bash
-        run: |
-          echo "::group::Create application archive"
-          tar -C ./dist/ -czf ./client.tgz .
-          echo "::endgroup::"
-      - name: Upload production-ready build files
-        uses: actions/upload-artifact@v4
-        with:
-          name: production-files
-          path: ./frontend/client.tgz
+          api-endpoint: https://reportvision-ocr-${{ inputs.deploy-env }}.azurewebsites.net/
+          frontend-tarball: ./frontend.tgz
```

Comment:
Is the npm packages being installed in the `./frontend.tgz` file?

Reply:
thats correct. at least with GHA, packaging all the frontend files into a tarball/zip makes downloading from other jobs smoother.